File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/GreaterThanRule.java
package com.example.navigation.stepsEngine.flow.rules.comparable;

        import com.example.navigation.stepsEngine.payment.FlowState;

public class GreaterThanRule extends ComparableRule {

    private Long count;

    public Long getCount() {
        return count;
    }

    public void setCount(Long count) {
        this.count = count;
    }

    @Override
    public boolean evaluate(FlowState flowState) {
        return (Long) getField().get(flowState) > count;
    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/GreaterThanRule.kt
package com.example.navigation.stepsEngine.flow.rules.comparable

import com.example.navigation.stepsEngine.payment.FlowState
import java.math.BigDecimal

class GreaterThanRule : ComparableRule() {
    lateinit var count: BigDecimal

    override fun evaluate(flowState: FlowState): Boolean {
        var value = getValue(flowState)
        return  (value as BigDecimal) > count
    }
}